### ISSUE TITLE
"Unexpected packet before version" using private key authentication

### ISSUE THREAD

**Issue by jradwan** [SchoofsKelvin/vscode-sshfs#185]
When trying to connect to an SSH file system using a private key for authentication, I get this error from the extension:

`Error while connecting to SSH FS <server name>: Unexpected packet before version`

I've tried all the different combinations:
- using the PuTTY/Pageant config
- setting the location of the .ppk file and prompting for the passphrase
- setting the location of the .ppk file and saving the passphrase in the extension settings

On the same server I have a second login with normal username/password authentication and that works fine with the extension. It just seems to be when trying to use the private key.

**Comment by SchoofsKelvin** [comment]
Might this be a similar issue to mscdex/ssh2-streams#121?

If it's a protocol issue, it's probably a bug in the libraries I use (ssh2 which in turn uses ssh2-streams) or a configuration/setup issue with your SSH server. Might also be related to #162 or #36, although this exact error message is unexpected.

**Comment by jradwan** [comment]
I figured it out. Turns out I wasn't just having this problem in VSCode but in _any_ sFTP client using my private key. The culprit turned out to be an environment script that was added to my .bashrc. The new script was generating output when executed, displayed to stdout. This was breaking the sFTP negotiation (I guess sFTP runs login scripts like a regular SSH connection). I edited .bashrc to send that script's output to /dev/null and the connection works now. Sorry for the false alarm. :)
